components/ws-manager-mk2/controllers/workspace_controller_test.go

```diff
@@ -362,8 +381,12 @@ func expectWorkspaceCleanup(ws *workspacev1.Workspace, pod *corev1.Pod) {
 	}, timeout, interval).Should(Succeed(), "pod did not go away")
 
 	By("cleaning up the workspace resource")
-	Eventually(func() error {
-		return checkNotFound(ws)
+	Eventually(func(g Gomega) error {
+		if err := checkNotFound(ws); err == nil {
+			return nil
+		}
+		g.Expect(ws.Status.Phase).To(Equal(workspacev1.WorkspacePhaseStopped))
```

Comment:
no change in behaviour, but this does now show what the workspace phase was last in if it doesn't get deleted. Easier to debug why a workspace doesn't get cleaned up